x-pack/plugins/actions/server/create_unsecured_execute_function.ts

```diff
@@ -0,0 +1,160 @@
+  consumer?: string;
+  relatedSavedObjects?: RelatedSavedObjects;
+}
+
+export type BulkUnsecuredExecutionEnqueuer<T> = (
+  internalSavedObjectsRepository: ISavedObjectsRepository,
+  actionsToExectute: ExecuteOptions[]
+) => Promise<T>;
+
+export function createBulkUnsecuredExecutionEnqueuerFunction({
+  taskManager,
+  connectorTypeRegistry,
+  isESOCanEncrypt,
+  preconfiguredConnectors,
+}: CreateBulkUnsecuredExecuteFunctionOptions): BulkUnsecuredExecutionEnqueuer<void> {
+  return async function execute(
+    internalSavedObjectsRepository: ISavedObjectsRepository,
+    actionsToExecute: ExecuteOptions[]
+  ) {
+    if (!isESOCanEncrypt) {
+      throw new Error(
+        `Unable to execute actions because the Encrypted Saved Objects plugin is missing encryption key. Please set xpack.encryptedSavedObjects.encryptionKey in the kibana.yml or use the bin/kibana-encryption-keys command.`
+      );
+    }
+
+    const connectorTypeIds: Record<string, string> = {};
+    const spaceIds: Record<string, string> = {};
+    const connectorIds = [...new Set(actionsToExecute.map((action) => action.id))];
+
+    const notPreconfiguredConnectors = connectorIds.filter(
+      (connectorId) =>
+        preconfiguredConnectors.find((connector) => connector.id === connectorId) == null
+    );
+
+    if (notPreconfiguredConnectors.length > 0) {
+      // log warning or throw error?
+    }
+
+    const connectors: PreconfiguredConnector[] = compact(
+      connectorIds.map((connectorId) =>
+        preconfiguredConnectors.find((pConnector) => pConnector.id === connectorId)
+      )
+    );
+
+    connectors.forEach((connector) => {
+      const { id, actionTypeId } = connector;
+      if (!connectorTypeRegistry.isActionExecutable(id, actionTypeId, { notifyUsage: true })) {
+        connectorTypeRegistry.ensureActionTypeEnabled(actionTypeId);
+      }
+
+      connectorTypeIds[id] = actionTypeId;
+    });
+
+    const actions = await Promise.all(
+      actionsToExecute.map(async (actionToExecute) => {
+        // Get saved object references from action ID and relatedSavedObjects
+        const { references, relatedSavedObjectWithRefs } = extractSavedObjectReferences(
+          actionToExecute.id,
+          true,
+          actionToExecute.relatedSavedObjects
+        );
+        const executionSourceReference = executionSourceAsSavedObjectReferences(
+          actionToExecute.source
+        );
+
+        const taskReferences = [];
+        if (executionSourceReference.references) {
+          taskReferences.push(...executionSourceReference.references);
+        }
+        if (references) {
+          taskReferences.push(...references);
+        }
+
+        spaceIds[actionToExecute.id] = actionToExecute.spaceId;
+
+        return {
+          type: ACTION_TASK_PARAMS_SAVED_OBJECT_TYPE,
+          attributes: {
+            actionId: actionToExecute.id,
+            params: actionToExecute.params,
+            apiKey: actionToExecute.apiKey,
```

Comment:
Since we are technically allowing for an apiKey, I think we'll need ESO support, which might block this on https://github.com/elastic/kibana/issues/133835.

Reply:
Removed apiKey parameter in https://github.com/elastic/kibana/pull/143282/commits/2fe5ce579c72935b0c2a57228292407e3e5c9050 and added allowlist for connector types (just email and slack for now but could be expanded to other connector types that don't need an API key